backend/app/services/sql_generation_adapter.py

```
@@ -113,16 +117,99 @@ class ConfiguredSQLGenerationAdapter(BaseModel):
+        if self.model is not None:
+            payload["model"] = self.model
+
+        http_request = Request(
+            f"{self.base_url.rstrip('/')}/generate-sql",
+            data=json.dumps(payload).encode("utf-8"),
+            headers={
+                "Accept": "application/json",
+                "Content-Type": "application/json",
+            },
+            method="POST",
+        )
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)